### PR TITLE
Add fix for objects with circular references causing maximum call stack size exceeded

### PR DESCRIPTION
**Pull request by SafalPandey**
* Add fix for objects with circular references
* Add test for `deepEqualExlcudingEvery` in case of circular keys

**Fixes:** #14

### CODE REVIEW

File: lib/chai-exclude.js [hunk 1]
```diff
@@ -12,12 +12,62 @@ module.exports = (chai, utils) => {
...
+
       return removeKeysFromObject(val, props, recursive)
     }
 
     return removeKeysFromArray(val, props, recursive)
   }
 
+  /**
+   * Get info on circular keys in an object.
+   * isCyclic      {Boolean} Boolean denoting whether the object has circular keys or not.
```
Comment (mesaugat) [inline lib/chai-exclude.js]: Outdated JSDoc

Comment (mesaugat) [reply lib/chai-exclude.js]: Also, a new line after the description.

**Comment by mesaugat** [comment]
@SafalPandey Thank you for the pull request. 💯

I'll take a look at this later.

File: lib/chai-exclude.js [hunk 2]
```diff
@@ -12,12 +12,66 @@ module.exports = (chai, utils) => {
    */
   function removeKeysFrom (val, props, recursive = false) {
     if (isObject(val)) {
+
+      // Stringify all circular object references.
+      // TODO: Think of a better way to stringify circular keys.
+      const cyclicInfo = getCyclic(val)
+      cyclicInfo.circularKeys.forEach((x) => eval(`val['${x.split('.').join("']['")}'] = '[Circular]'`))
```
Comment (mesaugat) [inline lib/chai-exclude.js]: I'm seeing `eval` after years. Also,`eval` is not considered safe. We might want to do something else here. Can't we mutate `val` without using `eval`?

Comment (SafalPandey) [reply lib/chai-exclude.js]: I was hesitant using `eval` as well. But I haven't found another way. I'm looking for an elegant way to do it but for now it works.

**Review by mesaugat — COMMENTED** [review]
@SafalPandey Instead of replacing circular references ourselves, I just used the [fclone](https://github.com/soyuka/fclone) package. 

This is just much faster and better. ¯\\_(ツ)_/¯

**Pull request merged**